### DISCUSSION TITLE
Unable to save to webpage after creating trigger according to CSS class

### DISCUSSION THREAD

**Discussion by jbrockc** [glacambre/firenvim#1571] [Q&A]
I decided to try out using the Nextcloud Mail app. I would like for Firenvim to work when I compose messages. The html for the compose window is:
```
<div data-v-18f7cd3a="" data-v-f6be8f7c="" name="body" class="message-body ck ck-content ck-editor__editable ck-rounded-corners ck-editor__editable_inline ck-blurred" lang="en" dir="ltr" role="textbox" aria-label="Rich Text Editor. Editing area: main" contenteditable="true">
<p class="ck-placeholder" data-placeholder="Write message …">
<br data-cke-filler="true">
</p>
</div>
```
In my `init.vim` I have the following:
```
let g:firenvim_config = {
    \ 'localSettings': {
	\ 'mynextclouddomain.com': {
	    \ 'selector': 'div[class="message-body"]',
	\ }
    \ }
\ }
```
This triggers Neovim when I go to compose. However, I cannot save what I write to the webpage, i.e. using `:x`. Also, Neovim does not reinitialize automatically after running `:q`. (Usually with my setup, if I run `:x` or `:q`, leave that page's focus, and then return again, Neovim reinitializes.)
I really appreciate this plugin. Thank you in advance for any help.

**Update:**
I recreated the html on a blank webpage with no Javascript, and I also put the full class attribute in my init.vim, i.e. "message-body ck ck-content etc". On my separate page, it worked fine. The Nextcloud mail composer is running some Javascript that is clearing any text written in Neovim. I am not sure how to remedy that from the Nextcloud side of things. I suppose I would just have to remember to do yG, then close, paste it, and then send my message. Any ideas are welcome. Thanks

**Comment by glacambre** [comment]
Unfortunately I believe there is indeed nothing you can do, pages that use a JS text editor have to have some extra support in Firenvim itself in order to call the JS APIs they use and directly modify the datastructures of the editor instead of the HTML.

**Reply by jbrockc** [reply]
I appreciate your reply!
